app/services/auth/session.server.ts

```diff
@@ -23,41 +27,34 @@ const { getSession, commitSession, destroySession } =
 
 export { commitSession, destroySession, getSession };
 
+interface CreateUserSessionProps extends AuthenticationContext {
+  request: Request;
+}
+
 // Once a user has authenticated with the OAuth2 strategy, we create a session for the user.
-export const createUserSession = async (
-  accessToken: string,
-  expiresAt: number,
-  refreshToken: string,
-  request: Request,
-) => {
+export const createUserSession = async ({
```

Comment:
I wonder if we could refactor these two functions. 

`createUserSession` and `updateUserSession` seem almost identical - one returns the result of `commitSession` while the other awaits it without returning.

Maybe it would be good to just merge them into a single `setSession`

Something like this:

```suggestion
interface SetSessionProps extends AuthenticationContext {
  request: Request;
}

export const setSession = async ({
  accessToken,
  expiresAt,
  refreshToken,
  request,
}: SetSessionProps) => {
  const session = await getSession(request.headers.get("Cookie"));
  session.set("accessToken", accessToken);
  session.set("expiresAt", expiresAt);
  session.set("refreshToken", refreshToken);

  console.log("setSession: accessToken is", accessToken);

  return await commitSession(session);
};
```

What do you think?